applications/nrf_desktop/src/modules/failsafe.c

```diff
@@ -55,17 +67,27 @@ static bool app_event_handler(const struct app_event_header *aeh)
 				cast_module_state_event(aeh);
 
 		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+			int err;
+			bool check_result;
 			static bool initialized;
 
 			__ASSERT_NO_MSG(!initialized);
 			initialized = true;
 
-			if (failsafe_check()) {
+			err = failsafe_check(&check_result);
+			if (!err && check_result) {
 				failsafe_erase();
 			}
-			failsafe_clear();
 
-			module_set_state(MODULE_STATE_READY);
+			if (!err) {
+				err = failsafe_clear();
```

Comment:
Consider avoiding clearing the reset reason if `failsafe_erase` call fails.

Reply:
This worked a bit differently in the previous implementation, where we cleared the reset reason unconditionally. Are you sure about that?

Reply:
I am aware that it used to work differently.
I am wondering a bit which approach would be better here: if we desist from clearing the reset reason on settings erase error, we give failsafe module another chance to erase settings which could be safer.